libs/libmysqlxx/CMakeLists.txt

```diff
@@ -26,25 +26,21 @@ add_library (mysqlxx
 
 add_dependencies (mysqlxx common)
 
-set (Z_HINTS "/usr/local/opt/zlib/lib")
-if (USE_STATIC_LIBRARIES)
-	find_library (Z_LIB libz.a HINTS ${Z_HINTS})
-else ()
-	find_library (Z_LIB z HINTS ${Z_HINTS})
-endif ()
-
 set(PLATFORM_LIBS ${CMAKE_DL_LIBS})
 
 if (USE_STATIC_LIBRARIES)
 	set (MYSQLCLIENT_LIB ${CMAKE_CURRENT_BINARY_DIR}/libmysqlclient.a)
-	target_link_libraries (mysqlxx common ${MYSQLCLIENT_LIB} ${OPENSSL_LIBRARIES} ${Z_LIB} ${PLATFORM_LIBS})
 	add_custom_command (
 		OUTPUT ${MYSQLCLIENT_LIB}
 		COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/patch.sh ${STATIC_MYSQLCLIENT_LIB} ${MYSQLCLIENT_LIB}
 		WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
 		COMMENT "Patching mysqlclient library.")
 	add_custom_target (our_mysql_client DEPENDS ${MYSQLCLIENT_LIB})
 	add_dependencies (mysqlxx our_mysql_client)
+	set (Z_LIB zlibstatic)
+else ()
+	set (Z_HINTS "/usr/local/opt/zlib/lib")
+	find_library (Z_LIB z HINTS ${Z_HINTS})
 endif ()
 
 target_link_libraries (mysqlxx common ${MYSQLCLIENT_LIB} ${OPENSSL_LIBRARIES} ${Z_LIB} ${PLATFORM_LIBS})
```

Comment:
Dynamic linking is not used at all for our builds.
It was added only by request of our users, who had convinced us, that they really need it.
Thus, disabling `USE_STATIC_LIBRARIES` option is loosely supported.

PS. If I understand correctly, dynamic linking will prefer symbols, that already exist in main executable (except the case, when dlopen with RTLD_DEEPBIND option is used). So, dynamic linking of libmysqlclient and static linking of libz should work fine.